ivy/array/extensions/elementwise.py

```diff
@@ -149,3 +149,53 @@ def fmax(
         ivy.array([ 0,  0,  nan])
         """
         return ivy.fmax(self._data, x2, out=out)
+
+    def trapz(
+        self: ivy.Array,
+        /,
+        *,
+        x: Optional[ivy.Array] = None,
+        dx: Optional[float] = 1.0,
+        axis: Optional[int] = -1,
+        out: Optional[ivy.Array] = None,
+    ) -> ivy.Array:
+        """ivy.Array instance method variant of ivy.trapz. This method simply
+        wraps the function, and so the docstring for ivy.trapz also applies to
+        this method with minimal changes.
+
+        Parameters
+        ----------
+        self
+            The array that should be integrated.
+        x
+            The sample points corresponding to the input array values. 
+            If x is None, the sample points are assumed to be evenly spaced
+            dx apart. The default is None.            
+        dx 
+            The spacing between sample points when x is None. The default is 1.
+        axis
+            The axis along which to integrate.
+        out
+            optional output array, for writing the result to.
+
+        Returns
+        -------
+        ret
+            Definite integral of n-dimensional array as approximated along
+            a single axis by the trapezoidal rule. If the input array is a
+            1-dimensional array, then the result is a float. If n is greater
+            than 1, then the result is an n-1 dimensional array.
+
+        Examples
+        --------
+        >>> y = ivy.array([1, 2, 3]) 
+        >>> ivy.trapz([1,2,3])
+        4.0
+        >>> y = ivy.array([1, 2, 3])
+        >>> ivy.trapz([1,2,3], x=[4, 6, 8])
```

Comment:
This example suggests `ivy.trapz` can accepts lists as input y... In that case the type hints need updating or maybe the example needs changing. Whichever is correct....